packages/core/src/interpreter.ts

```diff
@@ -345,8 +345,12 @@ export class Interpreter<
         listener(errorEvent.data.data);
       });
     } else {
-      this.stop();
-      throw errorEvent.data.data;
+      if (this.parent) {
+        this.parent?.send(errorEvent);
```

Comment:
you have just checked if there is a parent so we can just access `send` method here directly - without using the optional chaining:
```suggestion
        this.parent.send(errorEvent);
```